src/main/kotlin/process/RDPSimplification.kt

```diff
@@ -0,0 +1,84 @@
+package process
+
+/*
+* The Ramer–Douglas–Peucker algorithm is a line simplification algorithm
+* for reducing the number of points used to define its shape.
+*
+* Wikipedia: https://en.wikipedia.org/wiki/Ramer%E2%80%93Douglas%E2%80%93Peucker_algorithm
+* Implementation reference: https://rosettacode.org/wiki/Ramer-Douglas-Peucker_line_simplification
+* */
+
+// As Utaformatix use Pair<Long, Double> to describe pitch point, so...
+typealias Point = Pair<Long, Double>
+
+private fun perpendicularDistance(pt: Point, lineStart: Point, lineEnd: Point): Double {
+    var dx = (lineEnd.first - lineStart.first).toDouble()
+    var dy = lineEnd.second - lineStart.second
+
+    // Normalize
+    val mag = kotlin.math.hypot(dx, dy)
+    if (mag > 0.0) { dx /= mag; dy /= mag }
+    val pvx = pt.first - lineStart.first
+    val pvy = pt.second - lineStart.second
+
+    // Get dot product (project pv onto normalized direction)
+    val pvdot = dx * pvx + dy * pvy
+
+    // Scale line direction vector and subtract it from pv
+    val ax = pvx - pvdot * dx
+    val ay = pvy - pvdot * dy
+
+    return kotlin.math.hypot(ax, ay)
+}
+
+/** The RDP(Ramer–Douglas–Peucker) simplification algorithm for line based shape.
```

Comment:
Very slight problem but JavaDoc should be like:
```
/**
* ...
* ...
*/
```

(don't write in the first line)